src/bayesdb.py

```diff
@@ -142,7 +144,7 @@ def execute(self, string, bindings=None):
         if bindings is None:
             bindings = ()
         if self.tracer:
-            self.tracer(string, bindings)
+            finish_thunk = self.tracer(string, bindings)
```

Comment:
This was before the parse so that it would be easy to see what bayeslite was working on when whatever was going wrong went wrong. By delaying it, we may be making debugging significantly harder. Consider having two kinds of tracers, a pre-tracer and a post-tracer, and leaving pre-tracers as-is? In any case, discuss design with riastradh-probcomp@